src/data/chats/model_loader.rs

```diff
@@ -1,88 +1,199 @@
-use std::{sync::mpsc::{channel, Receiver, Sender}, thread};
-use anyhow::Result;
+use anyhow::anyhow;
 use makepad_widgets::SignalToUI;
-use moxin_backend::Backend;
-use moxin_protocol::{data::File, protocol::{Command, LoadModelOptions, LoadModelResponse}};
-
-pub struct ModelLoader {
-    pub complete: bool,
-    pub file: File,
-    load_sender: Sender<Result<()>>,
-    load_receiver: Receiver<Result<()>>,
+use moxin_protocol::{
+    data::FileID,
+    protocol::{Command, LoadModelOptions, LoadModelResponse},
+};
+use std::{
+    sync::{
+        mpsc::{channel, Receiver, Sender},
+        Arc, Mutex,
+    },
+    thread,
+};
+
+/// Immutable handleable error type for the model loader.
+///
+/// Actually, just a wrapper around `anyhow::Error` to support `Clone`.
+#[derive(Debug, Clone)]
+pub struct ModelLoaderError(Arc<anyhow::Error>);
```

Comment:
Anyhow error is not clone. If we want to send an error as a result and save it in the `Failed` status at the same time, we need to wrap it. And since this is pretty generic it could be a `MoxinError` actually.

But this is unnecessary, as we are not reading the error from the status currently. We are just reading the error in the result of `load_blocking`. So we actually don't need to be clone. So we can just remove this.

Unless you believe this can be useful. If not I will remove the error from `Failed` and this wrapper with it.